### ISSUE TITLE
comment on exported function F should be of the form "F ..."

### ISSUE THREAD

**Issue by andrebolle** [golang/lint#488]
I was mislead by this lint info

**comment on exported function F should be of the form "F  ..."**

I tried "F is cool"
Then I tried // "F is cool"

I think I might not have been mislead if the lint info were worded as follows:

**The comment on the exported function F must be of the form "// F ..."**

While I'm here, may I suggest the follow lint message:

**exported function F should have a comment or be unexported**

be changed to

**The exported function F must have a comment of the form "// F ..."**

And why are the definite and indefinite articles (the, a, an) missing?

Love and Peace,
Andre

**Comment by hugolgst** [comment]
Hello Andre,

The comment should be of the form: 
```golang
// Foo returns foo
func Foo() string {
    return "foo"
}
```

I think this is pretty clear for me, does it seems unclear to other people?

**Comment by purelazy** [comment]
It's the quotes (") that threw me off. Maybe the editor could add the comment automatically

// F you should describe me here
func F()

After all, no-one likes boilerplate.

**Comment by nishakm** [comment]
Hello. This was actually not clear to me. May I suggest changing it to "comment on exported function F should start with the function name F"?

**Comment by MihaiPopescu1985** [comment]
The message should include the "returns" word at least.
How it is now:
`comment on exported function Foo should be of the form "Foo ..."`
How it must be:
`comment on exported function Foo should be of the form: // Foo returns ...`

**Comment by myfatemi04** [comment]
Golint should support whitespace before the comment, it looks weird when there are multiple lines
For example, this gives me a linting error:
```go
/* KTree stores a (...)
 */
type KTree struct {
	...
}
```
While this doesn't:
```go
/*KTree stores a (...)
 */
type KTree struct {
	...
}
```
Which looks a little weird in my opinion

**Comment by pezaremba** [comment]
For people who get here the same way I did: 

In my case, my habit is to use /**/ comments and I still received this error.

You **must** use // comments to get rid of this warning.

**Comment by myfatemi04** [comment]
@pezaremba It has worked for me with `/**/` comments if I start on the next line

**Comment by zargold** [comment]
Ok so in my case:

I had something following a JavaDoc string/JavaScript docstring:

```golang
/**
* IsComparable: checks if something is comparable
* @input: value
* @return: boolean: true if is comparable or false otherwise
*/
```
And this failed: is that simply because I had a colon as part of the function name (if so that seems a bit brittle?)

**Comment by andrebolle** [comment]
This issue doesn't have much traction. So I'll close it if I can.

**Comment by zargold** [comment]
I don't know what you mean... I'm still actually having the problem even after removing the colon... I really don't understand how my comment right above my function fails to pass validation. Were you able to resolve your hound complaint?

```go
/**
* CheckThing is where we determine
*	 if thing is blue
*
* @input: thing
* @return: nil or an error (if thing was blue
 */
func CheckThing(thing item.Specs) error {
```

**Comment by ianlancetaylor** [comment]
Go code doesn't normally use the

```
/**
 *
```

style.  I'm not sure, but I would guess that lint thinks that the comment is `* * CheckThing...` which does not start with `CheckThing`.
